airbyte-integrations/connectors/destination-kvdb/destination_kvdb/client.py

```diff
@@ -0,0 +1,98 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from typing import Any, Iterable, List, Mapping, Tuple, Union
+
+import requests
+
+
+class KvDbClient:
+    base_url = "https://kvdb.io"
+    PAGE_SIZE = 1000
+
+    def __init__(self, bucket_id: str, secret_key: str = None):
+        self.secret_key = secret_key
+        self.bucket_id = bucket_id
+
+    def write(self, key: str, value: Mapping[str, Any]):
+        return self.batch_write([(key, value)])
+
+    def batch_write(self, keys_and_values: List[Tuple[str, Mapping[str, Any]]]):
+        """
+        https://kvdb.io/docs/api/#execute-transaction
+        """
+        request_body = {"txn": [{"set": key, "value": value} for key, value in keys_and_values]}
+        return self._request("POST", json=request_body)
+
+    def list_keys(self, list_values: bool = False, prefix: str = None) -> Iterable[Union[str, List]]:
+        """
+        https://kvdb.io/docs/api/#list-keys
+        """
+        # TODO handle rate limiting
```

Comment:
I've been seeing a bunch of rate limiting rough edges we would immediately solve by being able to specify QPS in the CDK. Should we create a ticket to track this? This can then be linked to that issue.

Reply:
https://github.com/airbytehq/airbyte/issues/2742